chpl/chpl-service/src/main/java/gov/healthit/chpl/util/Util.java

```diff
@@ -112,64 +105,18 @@ public static Date getNewDate(final Date orig) {
     }
 
     public static int sortCriteria(CertificationCriterionDTO c1, CertificationCriterionDTO c2) {
-        Pattern pattern = Pattern.compile(
-                "^(\\d{3}\\.\\d{3})\\s{1}"      // captures "170.314"
-                        + "(\\([a-z]{1}\\))"    // captures "(b)"
-                        + "(\\([0-9]{1,2}\\))?" // captures "(5)" or "(12)"
-                        + "(\\([A-Z]{1}\\))?$"  // captures "(A)"
-                );
-        Matcher m1 = pattern.matcher(c1.getNumber());
-        Matcher m2 = pattern.matcher(c2.getNumber());
-        int ret = 0;
-        if (!m1.matches() || !m2.matches()) {
-            return ret;
-        }
-        ret = compareStrings(m1.group(NUMBER_TITLE), m2.group(NUMBER_TITLE));
-        if (ret != 0) {
-            return ret;
-        }
-        ret = compareStrings(m1.group(NUMBER_PARA_1), m2.group(NUMBER_PARA_1));
-        if (ret != 0) {
-            return ret;
-        }
-
-        ret = compareInts(m1.group(NUMBER_PARA_2), m2.group(NUMBER_PARA_2));
-        if (ret != 0) {
-            return ret;
-        }
-        ret = compareStrings(m1.group(NUMBER_PARA_3), m2.group(NUMBER_PARA_3));
-        if (ret != 0) {
-            return ret;
-        }
-        return c1.getTitle().compareTo(c2.getTitle());
+        String valueA = Util.formatCriteriaNumber(c1);
+        String valueB = Util.formatCriteriaNumber(c2);
+        return Util.getCertificationResultSortIndex(valueA) - Util.getCertificationResultSortIndex(valueB);
     }
 
-    private static int compareStrings(String s1, String s2) {
-        if (s1 == null && s2 == null) {
-            return 0;
-        }
-        if (s1 == null) {
-            return -1;
+    private static Integer getCertificationResultSortIndex(String criteriaNumber) {
+        Integer index = certificationResultSortOrder().indexOf(criteriaNumber);
+        if (index.equals(-1)) {
+            // This is case when the criteria number is not in the array, just make it last...
+            index = 10000;
```

Comment:
This is a small thing and I realize the code will work as written but could you make it like Integer.MAX_VALUE instead of a randomly chosen large number?